sdt_dask/dataplugs/README.md

```diff
@@ -0,0 +1,80 @@
+
+Example structure:
+
+```python
+import pandas as pd
+from sdt_dask.dataplugs.dataplug import DataPlug
+
+class YourDataPlug(DataPlug):
+    def __init__(self, param1, param2):
+        # Initialization code here
+
+    def get_data(self, keys: tuple) -> pd.DataFrame:
+        # Data retrieval and cleaning code here
+        return df
+```
+
+## Existing DataPlug Examples
+
+Below are detailed descriptions of the DataPlugs available for use with the SDT Dask tool. Each DataPlug serves a unique purpose, designed to retrieve and process data from different sources. A corresponding requirements.txt file for each DataPlug is located in the directory solar-data-tools/sdt_dask/dataplugs/Requirements/. 
+
+### 1. LocalFiles DataPlug (dataplugs/csv_plug.py)
+
+- **Description**: Retrieving and cleaning solar data from local CSV files
+- **Initialization**:   
+`path_to_files`: The directory path where the CSV files are stored.  
+- **`get_data` Tuple input**: Expects a single string value representing the filename (without extension) of the dataset to be processed. Example to call `get_data` method:  
+```python 
+data_plug.get_data(("filename",))
+```
+
+### 2. PVDAQPlug (dataplugs/pvdaq_plug.py)
+
+- **Description**: Retrieving and cleaning solar data from the PVDAQ database
+- **Initialization**:   
+`api_key`: Your API key for accessing the PVDAQ data.
+- **`get_data` Tuple input**: Expects a tuple containing the site ID (integer) and the year (integer) for which data is to be retrieved. Example to call get_data method:
+```python 
+data_plug.get_data((site_id, year))
+```
+
+
+### 3. PVDBPlug (dataplugs/pvdb_plug.py)
+
+- **Description**: Retrieving and cleaning solar data from the PVDB (Redshift) database
+- **Initialization**: No input but assumes the API key is set as an environment variable REDSHIFT_API_KEY.
+- **`get_data` Tuple input**: Expects a tuple containing the site ID (string) and the sensor type (integer), identifying the specific dataset to be retrieved. Example to call get_data method:
+```python 
+data_plug.get_data(("site_id", sensor_type))
+```
+
+### 4. S3Bucket DataPlug
+
+- **Description**: Retrieving and cleaning solar data from S3 Bucket. And provides a function to get the full key list inside the given bucket name.
+- **Initialization**:  
+`api_key`: Your API key for accessing the PVDAQ data.  
+And also assumes the AWS configuration has been set up in local environment
+- **`get_data` Tuple input**: Expects a single string value in the tuple, specifying the key of the file in the bucket. Example to call get_data method:
+```python 
+data_plug.get_data(("s3-file-key",))
+```
+- **Important Notice 1.**: As mentioned above, botocore.client.S3 object include open connections and internal state, they cannot be correctly serialized and safely transferred across workers. Thus, we should initialize the client instance for each get_data requests. In general, it applies to all non-serializable objects.
```

Comment:
Combine two notices and say for thread safety and proper serialization, we must first create a boto3 session then create a client within that session. The session management is to ensure thread safety and the client is to ensure serializability (we make the client on the worker so that we don't have to ever serialize it).